sdk-extensions/metric-incubator/src/test/java/io/opentelemetry/sdk/viewconfig/ViewConfigTest.java

```diff
@@ -0,0 +1,214 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.viewconfig;
+
+import static io.opentelemetry.sdk.testing.assertj.OpenTelemetryAssertions.assertThat;
+import static org.assertj.core.api.Assertions.as;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import io.opentelemetry.api.common.AttributeKey;
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.sdk.autoconfigure.spi.ConfigurationException;
+import io.opentelemetry.sdk.metrics.SdkMeterProvider;
+import io.opentelemetry.sdk.metrics.SdkMeterProviderBuilder;
+import io.opentelemetry.sdk.metrics.common.InstrumentType;
+import io.opentelemetry.sdk.metrics.internal.view.ViewRegistryBuilder;
+import io.opentelemetry.sdk.metrics.view.Aggregation;
+import io.opentelemetry.sdk.metrics.view.InstrumentSelector;
+import io.opentelemetry.sdk.metrics.view.View;
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.util.Arrays;
+import java.util.List;
+import java.util.UUID;
+import java.util.function.Consumer;
+import org.assertj.core.api.InstanceOfAssertFactories;
+import org.junit.jupiter.api.Test;
+
+class ViewConfigTest {
+
+  @Test
+  void registerViews_InvalidFile() {
+    assertThatThrownBy(
+            () ->
+                ViewConfig.registerViews(
+                    SdkMeterProvider.builder(), new File("/" + UUID.randomUUID())))
+        .isInstanceOf(ConfigurationException.class)
+        .hasMessageContaining("An error occurred reading view config file:");
+  }
+
+  @Test
+  void registerViews_FileFullConfig() {
+    SdkMeterProviderBuilder builder = SdkMeterProvider.builder();
+
+    ViewConfig.registerViews(
+        builder, new File(ViewConfigTest.class.getResource("/full-config.yaml").getFile()));
```

Comment:
While in practice we don't expect `getFile` to fail when running tests it always makes me nervous calling it on a resource, how about moving the buffered reader stuff to `registerViews` so we can pass an `InputStream` in both from the SPI and tests?

Reply:
Alternatively looks like you have a `bufferedReader` method in this class, but could still consider moving that into ViewConfig

Reply:
If `registerViews` only accepts an `InputStream`, the API surface area becomes really simple and folks can use it however they want - whether that be from a string, file, etc. I like it.